pasqal_cloud/batch.py

```diff
@@ -66,6 +69,36 @@ class Config:
         extra = Extra.allow
         arbitrary_types_allowed = True
 
+    @root_validator(pre=True)
+    def _build_ordered_jobs(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        """This root validator will modify the 'jobs' attribute which is a list
+        of jobs dictionaries ordered by creation time before instantiation.
+        It will duplicate the value of 'jobs' in a new attribute 'ordered_jobs'
+        to keep the jobs ordered by creation time.
+        """
+        ordered_jobs_list = []
+
+        jobs_received = values.get("jobs", [])
+
+        for job in jobs_received:
+            job_dict = {**job, "_client": values["_client"]}
+            ordered_jobs_list.append(job_dict)
+
+        values["ordered_jobs"] = ordered_jobs_list
+        return values
+
+    @property
+    def jobs(self) -> Dict[str, Job]:
+        """Once the 'ordered_jobs' is build, we need to keep the 'jobs' attribute
+        for backward compatibility with the code written by the users of the sdk
+        """
+        warn(
+            "'jobs' attribute is deprecated, use 'ordered_jobs' instead",
+            DeprecationWarning,
+            stacklevel=1,
+        )
+        return {**{job.id: job for job in self.ordered_jobs}}
+
```

Comment:
Should I add a setter ?

Reply:
It depends - is `add_job(...)` the only supported way of adding jobs or have we, in the past, accepted adding by adding to the dict?

Reply:
`add_job` should be the only entrypoint

Reply:
okay so no setter needed

Reply:
Actually the setter would be an option to get rid of the root validator @penguinoneshaw  no ?

Reply:
From all the codes I tried, pydantic V1 will prevent use from using setters, so wre will keep the root validator I think